### PR TITLE
HSEARCH-4925 Improve documentation of Elasticsearch/OpenSearch compatibility

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4925

### CODE REVIEW

**Comment by yrodiere** [comment]
I just started a CI run on `default|aws`: https://ci.hibernate.org/job/hibernate-search/job/PR-3671/2/

File: backend/elasticsearch-aws/pom.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
     <artifactId>hibernate-search-backend-elasticsearch-aws</artifactId>
 
     <name>Hibernate Search Backend - Elasticsearch - AWS integration</name>
-    <description>Configurer for the Hibernate Search Elasticsearch backend enabling connection to an AWS-hosted Elasticsearch cluster</description>
+    <description>Configurer for the Hibernate Search Elasticsearch backend enabling connection to an AWS-hosted Elasticsearch/OpenSearch cluster</description>
```
Comment (marko-bekhta) [inline backend/elasticsearch-aws/pom.xml:20]: not related to this PR 😃 
It makes sense to mention that this lib works with ES OS in its description, but I  was wondering ... we have a lot of mentions of ES or ES cluster throughout the docs, while it is kinda similar in that it is either ES or OS cluster ... Should we somehow "combine" the two into a Search Services cluster or something like that, or... or leave it as it is 😄

File: documentation/src/main/asciidoc/public/getting-started/orm/index.adoc [hunk 2]
```diff
@@ -80,7 +80,7 @@ which allows indexing and searching these documents.
 Below are the most common setups and matching dependencies for a quick start;
 read xref:{reference-documentation-location}#architecture[Architecture] for more information.
 
-Hibernate ORM + Lucene::
+Hibernate ORM + xref:{reference-documentation-location}#backend-lucene[Lucene]::
```
Comment (marko-bekhta) [inline documentation/src/main/asciidoc/public/getting-started/orm/index.adoc:83]: maybe let's do the same for the standalone mapper as well ? ( ` Standalone POJO Mapper + Lucene:: `)

**Review by marko-bekhta — APPROVED** [review]
looks nice 😃

File: backend/elasticsearch-aws/pom.xml (hunk 1 above)
Comment (yrodiere) [reply backend/elasticsearch-aws/pom.xml:20]: Yeah I don't have a proper solution for that... "Search Services" is too generic and IMO confusing.

The only actual solutions involving specific language would require either to use "Elasticsearch/OpenSearch" everywhere, or to simply duplicate the backend and documentation. Neither seems particularly better than the status quo...

File: documentation/src/main/asciidoc/public/getting-started/orm/index.adoc (hunk 2 above)
Comment (yrodiere) [reply documentation/src/main/asciidoc/public/getting-started/orm/index.adoc:83]: Right, will do

File: documentation/src/main/asciidoc/public/reference/_backend-elasticsearch.adoc [hunk 3]
```diff
@@ -182,9 +182,9 @@ your password will be transmitted in clear text over the network.
 === [[elasticsearch-integration-configuration-aws]] Authentication on Amazon Web Services
 
 The Hibernate Search Elasticsearch backend, once configured, will work just fine in most setups.
-However, if you need to use Amazon's https://docs.aws.amazon.com/elasticsearch-service/[managed Elasticsearch service],
+However, if you need to use Amazon's link:{amazonOpenSearchServiceUrl}[managed OpenSearch service],
 you will find it requires a proprietary authentication method:
-https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-request-signing.html[request signing].
+https://docs.aws.amazon.com/opensearch-service/latest/developerguide/request-signing.html[request signing].
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply documentation/src/main/asciidoc/public/reference/_backend-elasticsearch.adoc:232]: Right, will do.

**Comment by yrodiere** [comment]
Thanks, I addressed your comments.

New build for `default|aws`: https://ci.hibernate.org/job/hibernate-search/job/PR-3671/4/

**Pull request merged**